packages/__docs__/src/ComponentTheme/index.tsx

```diff
@@ -24,29 +24,20 @@
 
 /** @jsx jsx */
 import { Component } from 'react'
-import PropTypes from 'prop-types'
-
 import { Table } from '@instructure/ui-table'
 import { View } from '@instructure/ui-view'
 
 import { ColorSwatch } from '../ColorSwatch'
 
 import { withStyle, jsx } from '@instructure/emotion'
 import generateStyle from './styles'
-
+import type { ComponentThemeProps } from './props'
+import { propTypes } from './props'
 @withStyle(generateStyle, null)
-class ComponentTheme extends Component {
-  static propTypes = {
-    componentTheme: PropTypes.object.isRequired,
-    themeVariables: PropTypes.object.isRequired,
-    // eslint-disable-next-line react/require-default-props
-    makeStyles: PropTypes.func,
-    // eslint-disable-next-line react/require-default-props
-    styles: PropTypes.object
-  }
-
-  mapColors(colorKey) {
-    const map = {}
+class ComponentTheme extends Component<ComponentThemeProps> {
+  static propTypes = propTypes
+  mapColors(colorKey: any) {
```

Comment:
We actually know that this will have a type of `BaseColors`. But because TS is not very good with typing `Object.keys().forEach()`, you will need to have a special type cast there too:
```js
  mapColors(colorKey: BaseColors) {
    const map: Record<string, keyof BaseColors> = {}
    ;(Object.keys(colorKey) as Array<keyof BaseColors>).forEach((color) => {
      const hex = colorKey[color]
      if (typeof map[hex] === 'undefined') {
        map[hex] = color
      }
    })
    return map
  }
```